### PR TITLE
preserve query and hash when adding url dep

### PR DESCRIPTION
**Pull request by jdanyow**
related to #612 

1. updates addURLDependency to preserve asset path's query and hash
2. updates urlJoin to preserve asset's query and hash instead of preserving the public url's query and hash.

### CODE REVIEW

File: src/Asset.js [hunk 1]
```diff
@@ -78,15 +78,24 @@ class Asset {
       from = this.name;
     }
```
Comment (brandon93s) [inline src/Asset.js]: Can be simplified if we go ahead and use `URL.parse()`.

```js
let resolved = path.resolve(path.dirname(from), url).replace(/[?#].*$/, ''); // same as today

const parsed = URL.parse(url);

return (
  this.options.parser
    .getAsset(resolved, this.package, this.options)
    .generateBundleName() + parsed.search + parsed.hash
);
```

Comment (jdanyow) [reply src/Asset.js]: sure- was overthinking the perf implications I guess.

Comment (jdanyow) [reply src/Asset.js]: before I go any further- what should we do about this:

HTMLAsset calls `addURLDependency`, followed by `urlJoin`:
https://github.com/parcel-bundler/parcel/blob/5c5d5f8af634c0e0aa8e8a3542892febe7c27e85/src/assets/HTMLAsset.js#L50-L55

`urlJoin` preserves the query and hash of the publicUrl, not the assetPath...
https://github.com/parcel-bundler/parcel/blob/5c5d5f8af634c0e0aa8e8a3542892febe7c27e85/src/utils/urlJoin.js#L8-L12

should be the other way around right?

Comment (jdanyow) [reply src/Asset.js]: (went ahead and updated the PR with the necessary changes to urlJoin)

Comment (devongovett) [reply src/Asset.js]: Yeah this seems kinda brittle. I think using URL.parse + URL.stringify would be better probably.

Comment (jdanyow) [reply src/Asset.js]: updated!

**Comment by devongovett** [comment]
Awesome, thanks!

**Pull request merged**